docs/adr/0004-bundling-experiments.md

```diff
@@ -0,0 +1,74 @@
+
+Moving forward, we are discussing **(C) Retrieve Experiment data on first run, and deal with delay** to address the first product experimentation requirement, which involves the ability to run experiments on a user's first run, but not necessarily early (The first experiments don't happen to involve onboarding)
+
+### Positive Consequences
+
+* Since the experiments are bundled with the app, they will always be available on first run, without having to wait on network I/O. This allows us to create experiments early on the user's first run (for example, onboarding experiments)
+
+### Negative Consequences
+
+* Inability to disable experiments without issuing a dot-release. The hinders our ability to quickly rollback any experiments that are causing problems. As a consequence, this will slow us down in that we would not be able to launch experiments and iterate quickly if it goes wrong
+* Inability to pause enrollments remotely. We currently have a mechanism to pause enrollments directly on experimenter, however, since the experiment data from remote-settings is not applied until the second run, users would still get enrolled in experiments (whose enrollment has been paused) on their first run.
+* Experiment creation will be tied to release cycles, any experiments that would like to target users on their first run would need to be created before a release is made. This removes flexibility in experiment creation and creates limitation on the timeline experiments can be created to be tied with release cycles (or would force us to create dot-releases for experiments)
+
+## Pros and Cons of the Options
+
+### Do nothing
+
+* Good, because it keeps the flexibility in experiment creation
+* Good, because disabling experiments can still done remotely for all experiments
+* Bad, because it doesn't address the main problem of exposing experiments to user on their first run
+
+### Bundle Experiment data with app on release
+* Good, because it allows us to run experiments early on a user's first run
+* Good, because it prevents us from having to wait for experiments, especially if a user has a slow network connection
+* Bad, because it ties experiment creation with release cycles
+* Bad, because it prevents us from disabling problematic first-run experiments without a dot release
+* Bad, because it prevents us from pausing enrollment on first-run experiments without a dot release
+
+### Retrieve Experiment data on first run, and deal with delay 
+
+* Good, because it enables us to retrieve experiments for users on their first run
+* Good, because it keeps the flexibility in experiment creation
+* Good, because disabling experiments can still done remotely for all experiments
+* Bad, because experiments may not be ready early on the user's experience
```

Comment:
Thanks @jhugman! for
> Bad, because it changes the programming model from Nimbus being an idempotent configuration store to configuration changing non-deterministically.

Although it's true, I think it's more "makes nimbus **more** non-deterministic on the first run". Because the `apply` happens in a background thread regardless - but we definitely are magnifying the window of non-idempotency (definitely not a word 🙃) on the first run.  This can be mitigated a little (by having the consumers wait until the data is ready) but that isn't always possible (we leave the choice to the consumer). It's not great, which is why we have our demos and https://docs.google.com/document/d/1X1hC3t5zC7-Rp0OPIoiUr_ueLOAI0ez_jqslaNzOHjY/edit

can you clarify 
> Bad, because the experimentation platform could force the app to add unchangeable user interface for the entire population. This itself may have an effect on key metrics.

Is this in reference to loading indicators/etc that the app may decide to use if they want to wait for experiment data?